### ISSUE TITLE
BindingPattern shouldn't allow optional

### ISSUE THREAD

**Issue by Dunqing** [oxc-project/oxc#2253]
Please check the playground, These cases should report a syntax error

https://oxc-project.github.io/oxc/playground/?code=3YCAAID6gICAgICAgICXYPzir2mHELJNP5ai%2BRRJpeC%2FNQpaxAyIIQDW3aGxwOFp8XLQlq5uBOSOyMfgKq6ev2ENr1%2FjDLFCrJTzzc96PycEillagwONy03dP7z%2BdcIPd9VMDu1dLhMAJ5Cu2XSP7d9%2FLb8AgA%3D%3D

In typescript, I get the syntax error

https://www.typescriptlang.org/play?target=99#code/MYewdgzgLgBAhgfhgXhgIgIJoFCkrAbQCMBdJVAtAITRN3GhgG8ZgYBfc51gLnQGE0HIA

**Comment by Dunqing** [comment]
But we need to allow optional in FormalParameters

```ts
function a(b?: string) {}
```
